src/audio/dai-zephyr.c

```diff
@@ -1151,6 +1151,9 @@ static int dai_comp_trigger_internal(struct dai_data *dd, struct comp_dev *dev,
 		break;
 	}
 
+	if (!ret)
+		return comp_set_state(dev, cmd);
```

Comment:
I'd like to understand how this currently works. From the commit description it sounds like either `dai_copy()` is unneeded (in all currently working configurations) or like the SOF is currently broken for everybody, which doesn't seem to be the case.

Reply:
hm, if too misleading I can change the commit description to clearly state that this happens on NXP platforms. Indeed, this isn't broken for everybody, just NXP. Can't really tell why this works on Intel platforms because I'm not familiar with all the components used by Intel but I'm assuming it has something to do with the copier component which seems to be some sort of wrapper over the dai component. @lgirdwood @kv2019i @ranj063 could you please help clarify this?

Reply:
Must be set elsewhere on Intel as the copy callback is needed. i.e. on DW DMA to clear DONE bits.

Reply:
> I'm assuming it has something to do with the copier component

@LaurentiuM1234 ah, you're right, it might indeed be related to the IPC4 vs. IPC3 difference. Should we make it IPC3-only then?

Reply:
> > I'm assuming it has something to do with the copier component
> 
> @LaurentiuM1234 ah, you're right, it might indeed be related to the IPC4 vs. IPC3 difference. Should we make it IPC3-only then?

I think this is the correct thing todo, i.e. we should make sure this is called in an IPC generic place.

Reply:
I think making this IPC3-specific is the easier approach. Making this IPC version-agnostic means special care must be taken inside the `copier` component since `dai-zephyr` component can now change state on trigger. IMO, the second solution is better since `dai-zephyr` is a component that can be used with both IPC versions and adding IPC-specific code is not ideal. Despite this, I'm fine with any of them.

Reply:
Can we add some commentary and an issue # to this ifdef, it looks like we need to track this and remove the ifdef once root caused.

Reply:
created https://github.com/thesofproject/sof/issues/8920 + added comment.